devito/types/dense.py

```diff
@@ -346,7 +346,7 @@ def _size_outhalo(self):
 
         sizes = tuple(Size(i, j) for i, j in zip(left, right))
 
-        if self._distributor.is_parallel and (any(left) > 0 or any(right)) > 0:
+        if self._distributor.is_parallel and (any(left) or any(right)):
```

Comment:
is this fixing a legacy Issue?

Reply:
Yeah, iirc, the original condition works, but in an entirely unintuitive and unintended manner. Look at the positions of the brackets